.github/workflows/build_backend.yml

```diff
@@ -71,30 +71,30 @@ jobs:
 
     - name: Setup environment
       run: |
-        sudo bash ./buildscripts/ci/backend/setup.sh
+        bash ./buildscripts/ci/backend/setup.sh
```

Comment:
I think the reason for `sudo` was that some script files don't have the executable bit set, so Bash or the OS refused to run them without `sudo`. Of course the real way to fix this would be to modify the permissions of those files.

```Bash
# Unix-like platforms: update permission of actual file
chmod +x path/to/file.sh
git add path/to/file.sh
git commit -m "Add execute permission`

# Windows: Filesystem has no concept of "execute permission", so just 
# update permission in the Git database to share with other developers.
git update-index --chmod=+x path/to/file.sh
git commit -m "Add execute permission"
```

To see which `.sh` files are currently lacking execute permission:
```Bash
git ls-files --stage -- '*.sh' | grep '^...644' # 755 = permission; 644 = no permission
```

Reply:
I doubt whether that is the case:
- We (almost) always use `bash ./script.sh` rather than just `./script.h`; in that case, execute permissions are not necessary, afaik
- As far as I can see, using `sudo` does not help when a file doesn't have execute permission.

Reply:
A more likely explanation seems to me:
1. Some script needed sudo access, for example to use `apt`; it was decided to put the sudo at the top of the "call hierarchy", rather than only where it was needed
2. That turned out to cause problems, because the script that run in the `sudo` call tree saw a different home directory than the other scripts of other steps; to solve that, someone decided to put `sudo` before every script invocation
3. When adding new actions, existing actions were used as a reference, so the usage of `sudo` was replicated everywhere

And I think that at some point, I deleted all `sudo`s in the macOS action, because they were causing problems; but I did not delete them yet for other actions because I was afraid to break things; but now I'm quite confident that this should work.

Reply:
Ah yes, I think I replaced `bash script.sh` with `./script.sh` somewhere and then had to provide execute permission. So it was `bash` rather than `sudo` that I removed.